Automattic/wp-calypso
# Make it possible to connect to a GMB location in 2 steps

This PR updates the GMB connection flow to use the new `POST /sites/%s/keyrings/%s` endpoint introduced in D13771 to update a site keyring.

This allows us to have a tri-state for a site
- Disconnected from Google My Business
- Connected to a Google My Business account
- Connected to a Google My Business location

### Testing Instructions
- Apply server patch D13771
- Boot this branch locally
- Navigate to Stats for a business site
- Connect with google account 1, do not select a location.
- Go back to Stats, you should still see the nudge
- Connect with google account 2, notice only the locations associated with this account show up
- Connect to a location
- Go back manually to http://calypso.localhost:3000/google-my-business/select-business-type/:site_slug
- Connect with google account 1, notice you are disconnected from your previous location and only the locations for account 1 are shown
- Go to Sharing
- Try connecting to GMB with google account 1 or 2, only the locations from one account or the other should show up
- Try connecting with an account with no locations, you should be redirected to the new account page

### Reviews
